packages/core/src/credential-provider.ts

```diff
@@ -27,8 +36,39 @@ export function isBBSPlusCredential(credential) {
   );
 }
 
+type importCredentialFromUriParams = {
+  uri: string;
+  wallet: IWallet;
+  didProvider: IDIDProvider;
+  getAuthCode: (authorizationURL: string) => Promise<string>;
+};
+
+export async function importCredentialFromURI({
+  uri,
+  wallet,
+  didProvider,
+  getAuthCode,
+}: importCredentialFromUriParams & {
+  wallet: IWallet;
+}) {
+  // TODO: unify the impl with the existing import credential flow
```

Comment:
todo now or later?

Reply:
For later https://dock-team.atlassian.net/browse/DCKM-581